### ISSUE TITLE
Tiles Tearing when using xy::tmx::Map

### ISSUE THREAD

**Issue by pmauldin** [fallahn/xygine#25]
Hello,

I've been using xygine for a game I'm making and first I just want to say thanks so much @fallahn for the work you've done. It's really amazing.

I ran into a problem today though: after following the documentation [here](https://github.com/fallahn/xygine/wiki/Tiled-tmx-map-parser), I have my TMX file being displayed on screen, and it's working great except for one major issue. While moving the screen around, lines (about 1 pixel wide as far as I can tell) often appear in between row/columns of tiles. Examples:
[After startup](http://i.imgur.com/Cqje5IX.png)
[After moving camera](http://i.imgur.com/zHtF4b1.png).

So far what I've tried is making sure that my player's position (and as such the scene's camera's center pos) is always floored. Didn't seem to help any though. I'm not really sure what else to try.

My implementation can be [found here](https://github.com/pmauldin/Shift/blob/master/Shift/src/states/GameState.cpp).

Any help is greatly appreciated!

**Comment by fallahn** [comment]
Hi! Thanks for giving xygine a try. I'm sorry you seem to be experiencing this bug. I'm a little confused, however (not because of anything you've done) - this is actually a [well documented](https://en.sfml-dev.org/forums/index.php?topic=15338.0) problem when using vertex arrays to draw tiles, but the xy map renderer was specifically designed using [this](http://the2dgame.com/index.php?page=articles&ref=ART9) technique to avoid it. Off the top of my head there are variances between graphics card drivers (for instance lines like [this](https://github.com/fallahn/xygine/blob/master/xygine/include/xygine/shaders/Tilemap.hpp#L75) needed to be added to the shader), perhaps there may be something in that. Do you get the same problem when running the tilemap demo in the example project? If you want to chat at all I'm frequently on the [sfml channel](http://chat.sfml-dev.org/) during european daytime hours (and when I'm not there's usually someone friendly on hand who may be able to help out :) )

**Comment by pmauldin** [comment]
Thanks for the quick response!

I have zero experience with shaders, so I'll have to spend some time looking into that.

In the meantime, I loaded up the TilemapDemo example and was able to reproduce the problem (somewhat): I didn't see tearing on the stock example, but when I had it load in my TMX file instead, I start to see the same tearing as in my OP. The only difference that I could think of is that platform.tmx uses 64x64 tiles, while my map uses 32x32 tiles. Do you think this could possibly be related?

Thanks again.

**Comment by fallahn** [comment]
OK I've created a sample 32x32px map (which I've pushed to the repo if you wish to try it) and I can't recreate what I'm seeing with your screenshot - so this may possibly be down to your graphics card (is it an intel by any chance?). HOWEVER I have noticed that the issue appears between the 'chunks' which make up a layer because of the reason I linked above - each chunk is a single vertex array, and the lines can appear where two or more chunks join. [This](https://github.com/fallahn/xygine/blob/master/xygine/src/tilemap/Map.cpp#L358) line defines the current maximum size of a chunk, increasing it will reduce any tearing caused by ajoining chunks as there will be fewer chunks per layer - setting it to the same size as the map should in theory eliminate it completely (as there will only be one chunk per layer). The maximum size per chunk is that of the maximum texture size supported by your graphics card (modern hardware should easily support 4096 square textures) which you can query with `sf::Texture::getMaximumSize()` . It may be worth experimenting with this - if it proves useful then I'll happily make the size of the chunk a variable to make it easier to pass in any size you like.

**Comment by pmauldin** [comment]
Thanks so much for your help.

Changing the chunk size didn't seem to have any effect, funnily enough. Though when I set the size to `f::Texture::getMaximumSize()`, I got an error about it not being supported. I have a GTX1070, so NVidia.

However, I loaded up thirty_two.txm in the example project and couldn't seem to reproduce the tearing. So what I did was scrap my project and start from the ground up and now the issue is gone! It's extremely strange and I'm still very curious as to what the actual issue was, but at least I seem to be in a good state for now.
